packages/zip-it-and-ship-it/src/runtimes/node/utils/cpio.ts

```diff
@@ -0,0 +1,336 @@
+      // `symlinks` map, which we'll later use to create the symlinks in the
+      // target directory. We can't do that right now because the target path
+      // may not have been copied over yet.
+      if (stat.isSymbolicLink()) {
+        const targetPath = await readLink(srcFile)
+
+        // Two symlinks may point at the same target path, so keep a list of symlinks to create.
+        symlinks.set(targetPath, (symlinks.get(targetPath) ?? new Set()).add(absoluteDestPath))
+
+        return
+      }
+
+      return copyFile(srcFile, absoluteDestPath)
+    },
+    { concurrency: COPY_FILE_CONCURRENCY },
+  )
+
+  await pMap(
+    [...symlinks.entries()],
+    async ([target, paths]) => {
+      for (const path of paths) {
+        await mkdir(dirname(path), { recursive: true })
+        await symlink(target, path)
+      }
+    },
+    {
+      concurrency: COPY_FILE_CONCURRENCY,
+    },
+  )
+
+  return { path: functionFolder, entryFilename }
+}
+
+const createCpioArchive = async function ({
+  aliases = new Map(),
+  basePath,
+  branch,
+  cache,
+  destFolder,
+  extension,
+  featureFlags,
+  filename,
+  mainFile,
+  moduleFormat,
+  rewrites,
+  runtimeAPIVersion,
+  srcFiles,
+}: // generator,
+CpioNodeParameters) {
+  const destPath = join(destFolder, `${basename(filename, extension)}.cpio`)
+  const { archive, output } = startCpio(destPath)
+
+  // There is a naming conflict with the entry file if one of the supporting
+  // files (i.e. not the main file) has the path that the entry file needs to
+  // take.
+  const hasEntryFileConflict = conflictsWithEntryFile(srcFiles, {
+    basePath,
+    extension,
+    featureFlags,
+    filename,
+    mainFile,
+    runtimeAPIVersion,
+  })
+
+  // We don't need an entry file if it would end up with the same path as the
+  // function's main file. Unless we have a file conflict and need to move everything into a subfolder
+  const needsEntryFile =
+    featureFlags.zisi_unique_entry_file ||
+    runtimeAPIVersion === 2 ||
+    hasEntryFileConflict ||
+    !isNamedLikeEntryFile(mainFile, { basePath, featureFlags, filename, runtimeAPIVersion })
+
+  // If there is a naming conflict, we move all user files (everything other
+  // than the entry file) to its own sub-directory.
+  const userNamespace = hasEntryFileConflict ? DEFAULT_USER_SUBDIRECTORY : ''
+
+  let entryFilename = `${basename(filename, extname(filename))}.js`
+  let bootstrapVersion: string | undefined
+
+  if (needsEntryFile) {
+    const entryFile = getEntryFile({
+      commonPrefix: basePath,
+      filename,
+      mainFile,
+      moduleFormat,
+      userNamespace,
+      featureFlags,
+      runtimeAPIVersion,
+    })
+
+    entryFilename = entryFile.filename
+
+    addEntryFileToCpio(archive, entryFile)
+  }
+  // const telemetryFile = getTelemetryFile(generator)
+
+  // if (featureFlags.zisi_add_instrumentation_loader === true) {
+  //   addEntryFileToCpio(archive, telemetryFile)
+  // }
```

Comment:
this should be uncommented

```suggestion
 const telemetryFile = getTelemetryFile(generator)

 if (featureFlags.zisi_add_instrumentation_loader === true) {
   addEntryFileToCpio(archive, telemetryFile)
 }
```